src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -27,14 +26,47 @@ internal partial class DefaultHubDispatcher<THub> : HubDispatcher<THub> where TH
         private readonly IHubContext<THub> _hubContext;
         private readonly ILogger<HubDispatcher<THub>> _logger;
         private readonly bool _enableDetailedErrors;
+        private readonly Func<HubInvocationContext, ValueTask<object>> _invokeMiddleware;
+        private readonly Func<HubLifetimeContext, Task> _onConnectedMiddleware;
+        private readonly Func<HubLifetimeContext, Exception, Task> _onDisconnectedMiddleware;
 
-        public DefaultHubDispatcher(IServiceScopeFactory serviceScopeFactory, IHubContext<THub> hubContext, bool enableDetailedErrors, ILogger<DefaultHubDispatcher<THub>> logger)
+        public DefaultHubDispatcher(IServiceScopeFactory serviceScopeFactory, IHubContext<THub> hubContext, bool enableDetailedErrors,
+            ILogger<DefaultHubDispatcher<THub>> logger, List<IHubFilter> hubFilters)
         {
             _serviceScopeFactory = serviceScopeFactory;
             _hubContext = hubContext;
             _enableDetailedErrors = enableDetailedErrors;
             _logger = logger;
             DiscoverHubMethods();
+
+            var count = hubFilters?.Count ?? 0;
+            if (count != 0)
+            {
+                _invokeMiddleware = (invocationContext) =>
+                {
+                    if (invocationContext.ObjectMethodExecutor != null)
+                    {
+                        return ExecuteMethod(invocationContext.ObjectMethodExecutor, invocationContext.Hub, invocationContext.Arguments);
+                    }
+                    return ExecuteMethod(invocationContext.HubMethod.Name, invocationContext.Hub, invocationContext.Arguments);
+                };
+
+                _onConnectedMiddleware = (context) => context.Hub.OnConnectedAsync();
+                _onDisconnectedMiddleware = (context, exception) => context.Hub.OnDisconnectedAsync(exception);
+
+                for (var i = 1; i <= count; i++)
+                {
+                    var resolvedFilter = hubFilters[count - i];
```

Comment:
Isn't this just counting up from 0?

Reply:
No, it's going from count - 1 to 0

Reply:
Could you just do that via the `for` loop though? Seems like it would be clearer.